components/setup/RepoSelection.tsx

```diff
@@ -31,47 +31,69 @@ function formatRepoListInSaveSetupArgsForm(repos: RepoIdentifier[], install_id:
 	return Array.from(setupArgsMap.values());
 }
 
-const RepoSelection = ({ repoProvider, installId, setIsRepoSelectionDone }: { repoProvider: RepoProvider, installId: string, setIsRepoSelectionDone: Function | null }) => {
+
+const RepoSelection = ({ repoProvider, installId, setIsRepoSelectionDone, isNewAccordion }:
+	{ repoProvider: RepoProvider, installId: string, setIsRepoSelectionDone: Function | null, isNewAccordion: boolean }) => {
 	const [selectedRepos, setSelectedRepos] = useState<RepoIdentifier[]>([]);
+	const [disableAllRepos, setDisableAllRepos] = useState<boolean>(false);
 	const [allRepos, setAllRepos] = useState<RepoIdentifier[]>([]);
 	const [isGetReposLoading, setIsGetReposLoading] = useState<boolean>(false);
 	const [isRepoSubmitButtonDisabled, setIsRepoSubmitButtonDisabled] = useState<boolean>(false);
 	const [error, setError] = useState<string>("");
+	const [submitButtonText, setSubmitButtonText] = useState<string>("Submit");
 
 	useEffect(() => {
 		setIsGetReposLoading(true);
-		axios.get<{ repoList: RepoIdentifier[] }>('/api/docs/getAllRepos')
-			.then((response) => {
-				return response.data.repoList;
-			})
-			.then((allRepos) => {
-				const providerReposForUser = allRepos.filter(repo => repo.repo_provider === repoProvider);
-				setAllRepos(providerReposForUser)
-				
-				// automatically check the repositories that are already installed by the user
-				axios.get<{repoList: RepoIdentifier[]}>('/api/docs/getInstalledRepos', { params: { topicId: installId, provider: repoProvider } })
+		if (isNewAccordion) {
+			axios.get<{ repoList: RepoIdentifier[] }>('/api/docs/getInstalledRepos', { params: { topicId: installId, provider: repoProvider } })
 				.then((response) => {
 					return response.data.repoList;
 				})
-				.then((repos) => {
-					const filteredRepos = providerReposForUser.filter(repo => repos.some(selectedRepo => 
-						selectedRepo.repo_provider === repo.repo_provider &&
-						selectedRepo.repo_owner === repo.repo_owner &&
-						selectedRepo.repo_name === repo.repo_name
-					)); // this is important because we are matching object references when we compare values in the checkbox
-					setSelectedRepos(filteredRepos);
+				.then((allRepos) => {
+					const providerReposForUser = allRepos.filter(repo => repo.repo_provider === repoProvider);
```

Comment:
This seems redundant (given that the `axios.get` function call is sending the name of the provider in params.